src/react/projects/spark-core-react/src/SprkCard/SprkCard.js

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import warning from 'warning';
+import SprkCardHighlightedHeader from './components/SprkCardHighlightedHeader/SprkCardHighlightedHeader';
+import SprkCardTeaser from './components/SprkCardTeaser/SprkCardTeaser';
+
+const SprkCard = props => {
+  const {
+    additionalClasses,
+    children,
+    highlightedHeaderConfig,
+    idString,
+    standout,
+    teaserConfig,
+  } = props;
+
+  const highlightedHeaderConfigCount = Object.keys(highlightedHeaderConfig)
+    .length;
+  const teaserConfigCount = Object.keys(teaserConfig).length;
+  const onlyHasHighlightedHeader =
+    highlightedHeaderConfigCount > 0 && teaserConfigCount <= 2;
+  const onlyHasTeaser =
+    teaserConfigCount > 2 && highlightedHeaderConfigCount <= 0;
+  const hasNoVariant =
+    highlightedHeaderConfigCount <= 0 && teaserConfigCount <= 2;
+
+  // Card can either only be base, teaser, or highlighted header
+  warning(
+    onlyHasTeaser || onlyHasHighlightedHeader || hasNoVariant,
+    'SprkCard: cannot have teaserConfig and highlightedHeaderConfig propTypes at the same time.',
+  );
+
+  const GetCardContent = () => {
+    if (onlyHasHighlightedHeader) {
+      return (
+        <SprkCardHighlightedHeader
+          highlightedHeaderConfig={highlightedHeaderConfig}
+        />
+      );
+    }
+
+    if (onlyHasTeaser) {
+      return <SprkCardTeaser teaserConfig={teaserConfig} />;
+    }
+
+    return <React.Fragment>{children}</React.Fragment>;
+  };
+
+  return (
+    <div
+      className={cx('sprk-c-Card sprk-o-Stack', additionalClasses, {
+        'sprk-c-Card--standout': standout,
+        'sprk-u-TextAlign--center': teaserConfig.mediaType === 'icon',
+      })}
+      data-id={idString}
+    >
+      <GetCardContent />
+    </div>
+  );
+};
+
+SprkCard.propTypes = {
+  additionalClasses: PropTypes.string,
+  children: PropTypes.node,
+  highlightedHeaderConfig: PropTypes.shape({
+    // Text inside of highlighted card
+    bodyText: PropTypes.string,
+    // Sub-headline of card
+    description: PropTypes.string,
+    // Headline of card
+    title: PropTypes.string,
+  }),
+  // ID
```

Comment:
I would label this `data-id`. Don't want another dev to see this and put it in an id field.